src/Model.elm

```diff
@@ -498,3 +504,88 @@ update msg model =
 
                 Err error ->
                     { model | notifications = [], errors = (errorText error) :: model.errors } ! []
+
+        NewWebsocketUserMessage message ->
+            case (Mastodon.decodeWebSocketMessage message) of
+                Mastodon.EventError error ->
+                    { model | errors = error :: model.errors } ! []
+
+                Mastodon.NotificationResult result ->
+                    case result of
+                        Ok notification ->
+                            let
+                                {-
+                                   Limitation of Elm where you can't reference
+                                   model.notifications inside a { model | …)
+                                -}
+                                oldNotifications =
+                                    model.notifications
+
+                                {-
+                                   @FIXME: we should add a function to `Mastodon`
+                                   with this typeSignature :
+                                   Notification -> List NotificationAggregate -> List NotificationAggregate
+                                -}
+                                notificationAggregate =
+                                    Mastodon.NotificationAggregate
+                                        notification.type_
+                                        notification.status
+                                        [ notification.account ]
+                                        notification.created_at
+                            in
+                                { model | notifications = notificationAggregate :: oldNotifications } ! []
+
+                        Err error ->
+                            { model | errors = error :: model.errors } ! []
+
+                Mastodon.StatusResult result ->
+                    case result of
+                        Ok status ->
+                            let
+                                {-
+                                   Limitation of Elm where you can't reference
+                                   model.notifications inside a { model | …)
+                                -}
+                                oldLocalTimeline =
+                                    model.userTimeline
+                            in
+                                { model | userTimeline = status :: oldLocalTimeline } ! []
+
+                        Err error ->
+                            { model | errors = error :: model.errors } ! []
+
+        NewWebsocketLocalMessage message ->
+            -- @TODO
```

Comment:
We should probably implement these in this patch before it lands, what do you think?